src/signature.c

```diff
@@ -636,6 +673,31 @@ static int p11prov_sig_operate(P11PROV_SIG_CTX *sigctx, unsigned char *sig,
         }
     }
 
+    if (sigctx->mechtype == CKM_RSA_PKCS && sigctx->digest != 0) {
```

Comment:
Wondering if this should be instead a `if (sigctx->digest != 0) { switch (sigctx->mechtype)` and whitelist the mechanisms where we know the proper padding is done, or `p11prov_sig_set_mechanism` can setup the p11 to match openssl.